app/models/user/social_actions.rb

```diff
@@ -49,6 +49,11 @@ def build_message(conversation, opts={})
   end
 
   def find_or_create_participation!(target)
```

Comment:
I think we need to change the name of this, it no longer just finds it. `update_or_create_participation!`?